### PR TITLE
Add devise user token auth setup

### PR DESCRIPTION
**Pull request by wongshington**
the most important endpoints to take note of with this change:

## User
- Sign-Up
  - `POST to: /api/v1/users`
    - required information: `name`, `email`, `date_of_birth`, `password`
    - currently a "confirmation email" is sent to the email used to sign up

## Session
- Sign In
  - `POST to: /api/v1/sign_in`
    - required information: `email`, `password`
    - upon signing in, the response includes a `uid`, `access-token`, `client`, and `expiry` that must be passed back and forth on EACH request to the API

### CODE REVIEW

**Review by cschambacher — APPROVED** [review]

File: bgb_api/Gemfile [hunk 1]
```diff
@@ -6,15 +6,27 @@ ruby '2.7.1'
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
 gem 'rails', '~> 6.0.3', '>= 6.0.3.4'
 # Use sqlite3 as the database for Active Record
-gem 'sqlite3'
+# gem 'sqlite3'
```
Comment (pschambacher) [inline bgb_api/Gemfile:9]: Just remove?

File: bgb_api/app/controllers/omniauth_controller.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class OmniauthController < ApplicationController
+
+  def facebook
+    puts "reach the omniauth facebook controller function"
```
Comment (pschambacher) [inline bgb_api/app/controllers/omniauth_controller.rb:4]: Remove for cleanup? 🧹

Comment (pschambacher) [reply bgb_api/app/controllers/omniauth_controller.rb:4]: or at least use `Rails.logger.debug`

File: bgb_api/app/controllers/omniauth_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      render json: {message: 'Error with Facebook Login'}, status: 404
+    end
+  end
+  
+  def failure
+    puts "failure"
+    render json: {error: "bad news bears"}, status: 401
+  end
+
+end
```
Comment (pschambacher) [inline bgb_api/app/controllers/omniauth_controller.rb:22]: Missing carriage return at EOL. Change your editor settings 👹

File: bgb_api/config/database.yml [hunk 4]
```diff
@@ -1,24 +1,40 @@
...
-#   gem install sqlite3
-#
-#   Ensure the SQLite 3 gem is defined in your Gemfile
-#   gem 'sqlite3'
-#
-default: &default
-  adapter: sqlite3
-  pool: <%= ENV.fetch("RAILS_MAX_THREADS") { 5 } %>
-  timeout: 5000
+# # SQLite version 3.x
```
Comment (pschambacher) [inline bgb_api/config/database.yml:1]: 🧹  Remove

**Review by pschambacher — APPROVED** [review]

**Pull request merged**